src/Lazy_Load_Script.php

```diff
@@ -65,15 +69,19 @@ public function print_script() {
 				element.sizes = element.dataset.sizes;
 			}
 		} );
-	} )();
-} else {
-	( function() {
-		var script = document.createElement( 'script' );
+	} else if ( ! document.querySelector( 'script#native-lazyload-fallback' ) ) {
+		script = document.createElement( 'script' );
+		script.id = 'native-lazyload-fallback';
 		script.type = 'text/javascript';
 		script.src = '<?php echo esc_js( $this->get_fallback_script_url() ); ?>';
 		script.defer = true;
 		document.body.appendChild( script );
-	} )();
+	}
+};
+if ( document.querySelector( '.lazy' ) ) {
```

Comment:
Would checking for the value of `document.readyState` be more readable here?

Example:

```suggestion
if ( document.readyState === 'complete' || document.readyState === 'interactive' ) {
```

Reply:
The idea here is not to wait until the entire DOM content is loaded. My thinking was to run this as soon as possible after getting all the images. This script is either printed in the head or footer (if in-page, clearly somebody's doing something wrong), so as soon as lazy images are there, we can safely assume it's running in the footer and can do its logic immediately.

Reply:
IMO, this could be a bit too clever. In reality, images can appear after the `wp_footer` action is fired (e.g. by other plugins). So using `DOMContentLoaded` is safest.

Reply:
Fair point. My only worry is that e.g. a heavy script comes later, is executed and therefore delays the process when images start to be lazy-loaded, harming UX for images immediately in the viewport.

Can we label this as tradeoff to live with, or should we potentially even cater for running the logic twice? At least if the `loading` attribute is supported, we could run it immediately, and then on `DOMContentLoaded`, if there are images left, we can run it again?

I might be overthinking this, but I wanna ensure we consider these tradeoffs.

Reply:
Maybe start with `DOMContentLoaded` for now and then revisit later. More work is going to be needed later anyway for applying lazy-loading to content dynamically added to the page (e.g. Ajax, infinite scroll). So there will need to be some facilitate made for re-running this logic. In the case of content being added to the page, `MutationObserver` should be used.